datafusion/src/physical_plan/union.rs

```diff
@@ -82,11 +92,18 @@ impl ExecutionPlan for UnionExec {
     }
 
     async fn execute(&self, mut partition: usize) -> Result<SendableRecordBatchStream> {
+        let baseline_metrics = BaselineMetrics::new(&self.metrics, partition);
+        // record the tiny amount of work done in this function so
+        // elapsed_compute is reported as non zero
+        let timer = baseline_metrics.elapsed_compute().timer();
+
         // find partition to execute
         for input in self.inputs.iter() {
             // Calculate whether partition belongs to the current partition
             if partition < input.output_partitioning().partition_count() {
-                return input.execute(partition).await;
+                let stream = input.execute(partition).await?;
+                drop(timer);
```

Comment:
Is this drop needed if the next line contains a `return`?

Reply:
The drop is needed to satisify the borrow checker: as sed, `let timer = baseline_metrics.elapsed_compute().timer();` has borrowed from baseline_metrics and rust won't allow baseline metrics to be given to ObserverdStream::new while borrowed.

An alternative is to clone the elapsed compute
```rust
        let elapsed_compute = baseline_metrics.elapsed_compute().clone();
        let timer = elapsed_compute.timer();
```
Which is done in several other parts of this PR. 🤔 

Perhaps that would be better to keep the code consistent

Reply:
in https://github.com/apache/arrow-datafusion/pull/1018/commits/c0b656cef8020f34edc08756c6ac368118ad1b53